library/std/src/os/solid/io.rs

```diff
@@ -0,0 +1,113 @@
+//! SOLID-specific extensions to general I/O primitives
```

Comment:
The raw fd types and traits are now available under `library/std/src/os/fd/mod.rs`, so no need to repeat them here.

(And that also includes the new `BorrowedFd` and `OwnedFd` and related traits, which are missing here.)

Reply:
It seems that the raw FD types under `library/std/src/os/fd/mod.rs` assumes POSIX-like FDs, i.e., there's a unified namespace for both normal files and sockets, and both can be closed by `libc::close`. This doesn't apply to this platform.